node_modules/oae-admin/adminlogin/adminlogin.html

```diff
@@ -7,60 +7,24 @@ <h2 class="text-center">__MSG__SIGN_IN_TO_THE_OAE_ADMIN_PANEL__</h2>
 </div>
 
 <div id="adminlogin-authentication-template"><!--
-    {var hasLocalAuth = enabledStrategies['local'] || enabledStrategies['ldap']}
-    {var hasExternalAuth = enabledStrategies['cas'] || enabledStrategies['facebook'] || enabledStrategies['google'] ||
-                           enabledStrategies['googleApps'] || enabledStrategies['shibboleth'] || enabledStrategies['twitter']}
-
-    {var hasSingleInstitutionalAuth = _.keys(enabledStrategies).length === 1 && _.contains(['cas', 'googleApps', 'shibboleth'], _.keys(enabledStrategies)[0])}
-
-    {macro externalAuth(id, strategy, title, icon)}
-        <li>
-            <form action="${strategy.url}" method="POST">
-                <div class="form-group">
-                    <input type="hidden" name="redirectUrl" value="${redirectUrl|encodeForHTMLAttribute}" />
-                    <button type="submit" class="btn btn-external-auth btn-${id}">
-                        {if icon}
-                            <i class="fa fa-${icon}"></i>
-                        {/if}
-                        {if strategy.name}
-                            {var strategyName = strategy.name}
-                            __MSG__SIGN_IN_WITH_STRATEGY__
-                        {else}
-                            ${title}
-                        {/if}
-                    </button>
-                </div>
-            </form>
-        </li>
-    {/macro}
-
-    {if hasExternalAuth}
+    {if authStrategyInfo.hasExternalAuth}
+        {var opts = new Object()}
```

Comment:
Since the opts are only relevant when `authStrategyInfo.hasExternalAuth` is true, and that condition is being handled inside the template, I'd prefer the conditional construction of the object to be in the template as well.